sale_order_warehouse_location/models/sale_order.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("partner_shipping_id")
+    def _compute_warehouse_id(self):
+        ret_vals = super()._compute_warehouse_id()
+        warehouses = self.env["stock.warehouse"].search(
+            [("company_id", "in", [False] + self.company_id.ids)]
+        )
+        for order in self:
+            if not (
+                order.partner_shipping_id
+                and (order.state in ["draft", "sent"] or not order.ids)
+            ):
+                continue
+            available_warehouses = warehouses.filtered(
```

Comment:
```suggestion
            available_warehouse = first(warehouses.filtered(
```

So, you can do:

`if available_warehouse and order.warehouse_id != available_warehouse`

Reply:
@manuelregidor could u replay? thanks.

Reply:
@rousseldenis Changed